### ISSUE TITLE
Add "snapshots size" prometheus metric

### ISSUE THREAD

**Issue by skriss** [vmware-tanzu/velero#1309]
xref #1059 #1077 

> Another metric which can be valuable for me (but maybe questionable for you, I can move it to another issue for discussion or 'reject'):
>
>[gauge] size of all PV snapshots (labeled with a schedule, volumeSnapshotLocation) - to track how much space do I use.

Splitting this request out into its own issue.

**Comment by cpanato** [comment]
@skriss I would like to try this one, as a newbie here, can you help me and guide where should i change/add, or some examples in the code.
thanks!

**Comment by skriss** [comment]
@cpanato it'd be great if you could help out with this! Let me look into it and see if I can offer some guidance.

**Comment by skriss** [comment]
I took a quick look here, and there are definitely some questions to figure out before we can implement this:

- what do we mean by snapshot size? there are a couple of possibilities here:
    -  use the size of the disk that was snapshotted -- so if it's a 100GB disk, we report that as the snapshot size, even if there was only 1GB of data on it. 
    - ask the storage provider for the snapshot size -- however, it looks like there is some inconsistency between providers here -- for example, AWS will only report the disk size, whereas GCE has both disk size and snapshot size.
- how do we get the stat we're looking for?
    - we could get PV size from the Kubernetes API, if that's what we're interested in
    - if we want to get it from the storage provider, we likely need to add a new VolumeSnapshotter plugin function
    - CSI snapshots have a size field that we could use

I'm not sure off the top of my head what the best answers are to all of these questions, we need to do some thinking and solicit feedback. @cpanato I'm thinking maybe this isn't the best issue to pick up for a new contributor, unless it's something you're particularly interested in :)

**Comment by cpanato** [comment]
@skriss I agree because I don't have much experience with Velero, however if we get the scope done and some walkthrough I can try that 😄

**Comment by hamidaria70** [comment]
sorry for reviving such an old issue, but we would like to see these metrics. is there any interest in implementing it?
